x-pack/plugins/security_solution/public/management/components/artifact_list_page/artifact_list_page.test.tsx

```diff
@@ -0,0 +1,919 @@
+          });
+        });
+
+        afterEach(() => {
+          if (releaseApiUpdateResponse) {
+            releaseApiUpdateResponse();
+          }
+        });
+
+        it('should disable all buttons while an update is in flight', () => {
+          expect((getByTestId('testPage-flyout-cancelButton') as HTMLButtonElement).disabled).toBe(
+            true
+          );
+          expect((getByTestId('testPage-flyout-submitButton') as HTMLButtonElement).disabled).toBe(
+            true
+          );
+        });
+
+        it('should display loading indicator on Submit while an update is in flight', () => {
+          expect(
+            getByTestId('testPage-flyout-submitButton').querySelector('.euiLoadingSpinner')
+          ).toBeTruthy();
+        });
+
+        it('should pass `disabled=true` to the Form component while an update is in flight', () => {
+          expect(getLastFormComponentProps().disabled).toBe(true);
+        });
+      });
+
+      describe('and submit is successful', () => {
+        beforeEach(async () => {
+          await renderAndWaitForFlyout();
+
+          act(() => {
+            userEvent.click(renderResult.getByTestId('testPage-flyout-submitButton'));
+          });
+
+          await act(async () => {
+            await waitFor(() => {
+              expect(renderResult.queryByTestId('testPage-flyout')).toBeNull();
+            });
+          });
+        });
+
+        it('should show a success toast', async () => {
+          expect(coreStart.notifications.toasts.addSuccess).toHaveBeenCalledWith(
+            '"some name" has been added.'
+          );
+        });
+
+        it('should clear the URL params', () => {
+          expect(location.search).toBe('');
+        });
+      });
+
+      describe('and submit fails', () => {
+        beforeEach(async () => {
+          const _renderAndWaitForFlyout = renderAndWaitForFlyout;
+
+          renderAndWaitForFlyout = async (...args) => {
+            mockedApi.responseProvider.trustedAppCreate.mockImplementation(() => {
+              throw new Error('oh oh. no good!');
+            });
+
+            await _renderAndWaitForFlyout(...args);
+
+            act(() => {
+              userEvent.click(renderResult.getByTestId('testPage-flyout-submitButton'));
+            });
+
+            await act(async () => {
+              await waitFor(() =>
+                expect(mockedApi.responseProvider.trustedAppCreate).toHaveBeenCalled()
+              );
+            });
+
+            return renderResult;
+          };
+        });
+
+        // FIXME:PT investigate test failure
+        // (I don't understand why its failing... All assertions are successful -- HELP!)
+        it.skip('should re-enable `Cancel` and `Submit` buttons', async () => {
+          await renderAndWaitForFlyout();
+
+          const cancelButtonDisabledState = (
+            renderResult.getByTestId('testPage-flyout-cancelButton') as HTMLButtonElement
```

Comment:
Did you tried with this?
```typescript
expect(renderResult.getByTestId('testPage-flyout-cancelButton')).toBeEnabled();
```

Reply:
I don't see that `.toBeEnabled()` in jest's list of assertions

Reply:
were you thinking maybe with Enzyme?

Reply:
No, it is 🙂  -> https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/management/pages/policy/view/artifacts/delete_modal/policy_artifacts_delete_modal.test.tsx#L58-L59

Reply:
Cool. Did not realize this was there. thanks for pointing that out. I corrected the tests to use it.